src/Symfony/Bundle/FrameworkBundle/KernelBrowser.php

```diff
@@ -104,6 +107,31 @@ public function enableReboot()
         $this->reboot = true;
     }
 
+    /**
+     * @param UserInterface $user
+     */
+    public function loginUser($user, string $firewallContext = 'main'): self
+    {
+        if (!class_exists(UserInterface::class)) {
+            throw new \LogicException(sprintf('"%s" requires symfony/security-core to be installed.', __METHOD__));
+        }
+
+        if (!$user instanceof UserInterface) {
+            throw new \LogicException(sprintf('The first argument of "%s" must be instance of "%s", "%s" provided.', __METHOD__, UserInterface::class, \is_object($user) ? \get_class($user) : \gettype($user)));
```

Comment:
I'm not sure if I need to use the `get_debug_type()` function here already, or if #35945 has to be merged first.